core/controllers/collection_editor.py

```diff
@@ -102,20 +104,40 @@ def get(self, collection_id):
         collection = collection_services.get_collection_by_id(
             collection_id, strict=False)
 
+        can_edit = (
+            bool(self.user_id) and
+            self.username not in config_domain.BANNED_USERNAMES.value and
+            rights_manager.Actor(self.user_id).can_edit(
+                feconf.ACTIVITY_TYPE_COLLECTION, collection_id))
+
+
+        user_permissions = rights_manager.get_collection_rights(
+            collection_id).to_dict()
+        owner_names = user_permissions['owner_names']
+        editor_names = user_permissions['editor_names']
+        viewer_names = user_permissions['viewer_names']
+
         self.values.update({
-            'can_edit': True,
+            'can_edit': can_edit,
+            'can_modify_roles': rights_manager.Actor(
+                self.user_id).can_modify_roles(
+                    feconf.ACTIVITY_TYPE_COLLECTION, collection_id),
             'can_unpublish': rights_manager.Actor(
                 self.user_id).can_unpublish(
                     feconf.ACTIVITY_TYPE_COLLECTION, collection_id),
             'collection_id': collection.id,
+            'collection_version': collection.version,
+            'editors': editor_names,
             'is_private': rights_manager.is_collection_private(collection_id),
+            'owners': owner_names,
             'nav_mode': feconf.NAV_MODE_CREATE,
             'title': collection.title,
             'SHOW_COLLECTION_NAVIGATION_TAB_HISTORY': (
                 feconf.SHOW_COLLECTION_NAVIGATION_TAB_HISTORY),
             'SHOW_COLLECTION_NAVIGATION_TAB_STATS': (
                 feconf.SHOW_COLLECTION_NAVIGATION_TAB_STATS),
             'TAG_REGEX': feconf.TAG_REGEX,
+            'viewers': viewer_names,
```

Comment:
Not sure we need "viewers" for collections -- it's just a sequence of explorations, there's not much to view. Maybe just stick with owners and editors for now?